### PR TITLE
Add Snapshot Deployment yamls

### LINKED ISSUE #424: Add CRD and deployment for csi snapshot
Add CRD and Deployment yaml's for CSI Snapshot

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind feature

**What happened**:
Snapshot Support Implemented by following PR in Cinder CSI.
 [csi] add snapshot support #344 
- We need to add CRD's to use the snapshot functionality, and deployment file for external-snapshotter
- Update Document and CI

### PR DESCRIPTION
**Pull request by adisky**
**What this PR does / why we need it**:
Add Snapshot Deployment yamls

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #424

**Special notes for your reviewer**:

**Release note**:

```release-note
```
`NONE`

### CODE REVIEW

**Comment by dims** [comment]
@adisky looks like this needs a rebase

**Comment by dims** [comment]
/hold

(for rebase)

File: manifests/cinder-csi-plugin/csi-snapshotter-cinderplugin.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      app: csi-snapshotter-cinder
+  template:
+    metadata:
+      labels:
+        app: csi-snapshotter-cinder
+    spec:
+      serviceAccount: csi-snapshotter
+      containers:
+        - name: csi-snapshotter
+          image: quay.io/k8scsi/csi-snapshotter:v0.4.0
```
Comment (ramineni) [inline manifests/cinder-csi-plugin/csi-snapshotter-cinderplugin.yaml]: shouldn't we use v1.0.1 release here which supports csi spec 1.0?

Comment (adisky) [reply manifests/cinder-csi-plugin/csi-snapshotter-cinderplugin.yaml]: yes, it was 1.0.1, dont know what happened while rebasing :), will check and update

**Pull request merged**